### PR TITLE
MSTestV2 VB Templates

### PR DESCRIPTION
**Pull request by jayaranigarg**

### CODE REVIEW

File: Templates/MSTestTemplates.sln [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
-VisualStudioVersion = 15.0.26009.0
+VisualStudioVersion = 15.0.26403.0
 MinimumVisualStudioVersion = 10.0.40219.1
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MSTestDesktopVSIX", "Desktop\MSTestDesktopVSIX.csproj", "{9E08BE4A-D3C8-4E8D-AFBE-FA571F878464}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MSTestDesktopVB", "VisualBasic\Desktop\MSTestDesktopVB.csproj", "{1AFFAE1C-BD70-4201-B8C8-3862F56E784B}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MSTestUWPVSIX", "UWP\MSTestUWPVSIX.csproj", "{11207E9D-1AE5-41B2-A04A-5F2BC1DCD5E7}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MSTestUWPVB", "VisualBasic\UWP\MSTestUWPVB.csproj", "{49460594-BA56-4348-82AF-7B517263630B}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MSTestDesktopVSIX", "CSharp\Desktop\MSTestDesktopVSIX.csproj", "{9E08BE4A-D3C8-4E8D-AFBE-FA571F878464}"
```
Comment (AbhitejJohn) [inline Templates/MSTestTemplates.sln]: nit: To be consistent with the above, should this be MSTestDesktopCsharp?

File: Templates/VisualBasic/Desktop/ProjectTemplates/VisualBasic/Test/AssemblyInfo.vb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+﻿Imports System
+Imports System.Reflection
+Imports System.Runtime.InteropServices
+
+' General Information about an assembly is controlled through the following 
+' set of attributes. Change these attribute values to modify the information
+' associated with an assembly.
+
+' Review the values of the assembly attributes
```
Comment (AbhitejJohn) [inline Templates/VisualBasic/Desktop/ProjectTemplates/VisualBasic/Test/AssemblyInfo.vb]: Do we need these comments?

File: Templates/VisualBasic/Desktop/ProjectTemplates/VisualBasic/Test/Application.myapp [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<MyApplicationData xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
```
Comment (AbhitejJohn) [inline Templates/VisualBasic/Desktop/ProjectTemplates/VisualBasic/Test/Application.myapp:2]: nit: rename file to denote a Desktop UT VB template.

File: Templates/VisualBasic/Desktop/ProjectTemplates/VisualBasic/Test/Resources.Designer.vb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+'     Changes to this file may cause incorrect behavior and will be lost if
+'     the code is regenerated.
+' </auto-generated>
+'------------------------------------------------------------------------------
+
+Option Strict On
+Option Explicit On
+
+
+Namespace My.Resources
```
Comment (AbhitejJohn) [inline Templates/VisualBasic/Desktop/ProjectTemplates/VisualBasic/Test/Resources.Designer.vb:15]: why o why does a UT template require a resource file?

File: Templates/VisualBasic/Desktop/ProjectTemplates/VisualBasic/Test/Settings.Designer.vb [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+'     Changes to this file may cause incorrect behavior and will be lost if
+'     the code is regenerated.
+' </auto-generated>
+'------------------------------------------------------------------------------
+
+Option Strict On
+Option Explicit On
+
+
+Namespace My
```
Comment (AbhitejJohn) [inline Templates/VisualBasic/Desktop/ProjectTemplates/VisualBasic/Test/Settings.Designer.vb:15]: Is this file needed?

File: Templates/VisualBasic/Desktop/ProjectTemplates/VisualBasic/Test/Settings.settings [hunk 6]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version='1.0' encoding='utf-8'?>
+<SettingsFile xmlns="http://schemas.microsoft.com/VisualStudio/2004/01/settings" CurrentProfile="(Default)" UseMySettingsClassName="true">
```
Comment (AbhitejJohn) [inline Templates/VisualBasic/Desktop/ProjectTemplates/VisualBasic/Test/Settings.settings:2]: this one too. Can we remove it?

File: Templates/VisualBasic/Desktop/source.extension.vsixmanifest [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<PackageManifest Version="2.0.0" xmlns="http://schemas.microsoft.com/developer/vsx-schema/2011" xmlns:d="http://schemas.microsoft.com/developer/vsx-schema-design/2011">
+  <Metadata>
+    <Identity Id="MSTestDesktopVB.Microsoft.c6c7fff6-20cb-405d-9ad4-a60a6d0c55d9" Version="15.0.26127.00" Language="en-US" Publisher="Microsoft" />
```
Comment (AbhitejJohn) [inline Templates/VisualBasic/Desktop/source.extension.vsixmanifest:4]: Please do get this reviewed by Pratap.

File: Templates/VisualBasic/UWP/ProjectTemplates/VisualBasic/Windows UAP/Properties/AssemblyInfo.vb [hunk 8]
```diff
@@ -0,0 +1,32 @@
+﻿Imports System
+Imports System.Reflection
+Imports System.Runtime.InteropServices
+
+' General Information about an assembly is controlled through the following 
+' set of attributes. Change these attribute values to modify the information
+' associated with an assembly.
+
```
Comment (AbhitejJohn) [inline Templates/VisualBasic/UWP/ProjectTemplates/VisualBasic/Windows UAP/Properties/AssemblyInfo.vb]: Comments.

File: Templates/VisualBasic/UWP/ProjectTemplates/VisualBasic/Windows UAP/Properties/UnitTestApp.rd.xml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    This file contains Runtime Directives used by .NET Native. The defaults here are suitable for most
+    developers. However, you can modify these parameters to modify the behavior of the .NET Native
+    optimizer.
+
+    Runtime Directives are documented at https://go.microsoft.com/fwlink/?LinkID=391919
+
+    To fully enable reflection for App1.MyClass and all of its public/private members
+    <Type Name="App1.MyClass" Dynamic="Required All"/>
+
+    To enable dynamic creation of the specific instantiation of AppClass<T> over System.Int32
```
Comment (AbhitejJohn) [inline Templates/VisualBasic/UWP/ProjectTemplates/VisualBasic/Windows UAP/Properties/UnitTestApp.rd.xml:11]: Do check what the C# UWP template does with these comments.

File: Templates/VisualBasic/UWP/ProjectTemplates/VisualBasic/Windows UAP/UnitTestApp1.vbproj [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+    <OptionCompare>Binary</OptionCompare>
+  </PropertyGroup>
+  <PropertyGroup>
+    <OptionStrict>Off</OptionStrict>
+  </PropertyGroup>
+  <PropertyGroup>
+    <OptionInfer>On</OptionInfer>
+  </PropertyGroup>
+  <Import Project="$(MSBuildExtensionsPath)\Microsoft\WindowsXaml\v$(VisualStudioVersion)\Microsoft.Windows.UI.Xaml.VisualBasic.targets" />
+  <!-- To modify your build process, add your task inside one of the targets below and uncomment it.
```
Comment (AbhitejJohn) [inline Templates/VisualBasic/UWP/ProjectTemplates/VisualBasic/Windows UAP/UnitTestApp1.vbproj:188]: Would this need localization too?

File: Templates/VisualBasic/UWP/source.extension.vsixmanifest [hunk 11]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<PackageManifest Version="2.0.0" xmlns="http://schemas.microsoft.com/developer/vsx-schema/2011" xmlns:d="http://schemas.microsoft.com/developer/vsx-schema-design/2011">
+  <Metadata>
+    <Identity Id="MSTestUWPVB.Microsoft.3dc1d5cd-bbfb-456f-965e-5b962ad063d1" Version="15.0.26127.00" Language="en-US" Publisher="Microsoft" />
+    <DisplayName>MSTest V2 Templates for VB (Universal Windows)</DisplayName>
```
Comment (AbhitejJohn) [inline Templates/VisualBasic/UWP/source.extension.vsixmanifest:5]: Review for this too please.

**Comment by AbhitejJohn** [comment]
Just putting this on the radar too: We should make the change of hiding the old templates along with this change when we push in so we don't have two.

File: Templates/VisualBasic/Desktop/ProjectTemplates/VisualBasic/Test/AssemblyInfo.vb (hunk 2 above)
Comment (jayaranigarg) [reply Templates/VisualBasic/Desktop/ProjectTemplates/VisualBasic/Test/AssemblyInfo.vb]: Removed.

File: Templates/MSTestTemplates.sln (hunk 1 above)
Comment (jayaranigarg) [reply Templates/MSTestTemplates.sln]: Done.

File: Templates/VisualBasic/UWP/ProjectTemplates/VisualBasic/Windows UAP/Properties/AssemblyInfo.vb (hunk 8 above)
Comment (jayaranigarg) [reply Templates/VisualBasic/UWP/ProjectTemplates/VisualBasic/Windows UAP/Properties/AssemblyInfo.vb]: Removed.

File: Templates/VisualBasic/UWP/ProjectTemplates/VisualBasic/Windows UAP/UnitTestApp1.vbproj (hunk 10 above)
Comment (jayaranigarg) [reply Templates/VisualBasic/UWP/ProjectTemplates/VisualBasic/Windows UAP/UnitTestApp1.vbproj:188]: No.

File: Templates/VisualBasic/Desktop/MSTestDesktopVB.csproj [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+    <Content Include="packages\MSTest.TestAdapter.1.1.17.nupkg">
+      <IncludeInVSIX>true</IncludeInVSIX>
+    </Content>
+    <Content Include="packages\MSTest.TestFramework.1.1.17.nupkg">
+      <IncludeInVSIX>true</IncludeInVSIX>
+    </Content>
+    <None Include="source.extension.vsixmanifest">
+      <SubType>Designer</SubType>
+    </None>
+    <Content Include="DesktopVB.xml" />
```
Comment (Faizan2304) [inline Templates/VisualBasic/Desktop/MSTestDesktopVB.csproj]: Do we need "DesktopVB.xml" file in vsix? as per my understanding you guys read it to create "VBDesktopUnitTestTemplate.zip" file.

**Comment by jayaranigarg** [comment]
#132
